backend/src/api/events.js

```diff
@@ -0,0 +1,196 @@
+const express = require('express')
+const mongodb = require('mongodb')
+const router = express.Router()
+const { errorWrap } = require('../middleware')
+const { Event } = require('../models')
+
+// get all events
+router.get(
+  '/',
+  errorWrap(async (req, res) => {
+    let events = await Event.find()
+    res.json({
+      code: 200,
+      message: '',
+      result: events,
+      success: true
+    })
+  })
+)
+
+// get one event
+router.get(
+  '/:event_id',
+  errorWrap(async (req, res) => {
+    const event = await Event.findById(req.params.event_id)
+    res.json({
+      code: 200,
+      message: '',
+      result: event,
+      success: true
+    })
+  })
+)
+
+// create a new event
+router.post(
+  '/',
+  errorWrap(async (req, res) => {
+    const data = req.body
+    let response = 'Event Created Successfully'
+    let code = 404
+    let success = false
+    let eventName = data.name
+    let eventDate = data.date
+    let eventStartTime = data.startTime
+    let eventEndTime = data.endTime
+    let eventLocation = data.location
+    let eventDescription = data.description
+
+    if (eventName == undefined) {
+      response = 'Invalid Event Name'
+    } else if (eventDate == undefined) {
+      response = 'Invalid Event Date'
+    } else if (eventStartTime == undefined) {
+      response = 'Invalid Start Time'
+    } else if (eventEndTime == undefined) {
+      response = 'Invalid End Time'
+    } else if (eventLocation == undefined) {
+      response = 'Invalid Location'
+    } else if (eventDescription == undefined) {
+      response = 'Invalid Description'
+    } else {
+      const newEvent = new Event({
+        name: eventName,
+        date: eventDate,
+        startTime: eventStartTime,
+        endTime: eventEndTime,
+        location: eventLocation,
+        description: eventDescription,
+        attendees: []
+      })
```

Comment:
Because of the [`errorWrap`](https://github.com/hack4impact-uiuc/h4i-recruitment/blob/master/backend/src/middleware/errorWrap.js) middleware, we don't actually need to account for every possible error case. Instead, we can assume that the input is as we want, and if it isn't, the middleware will catch the exception and send the error message. I can explain more about middleware if you aren't familiar. I forgot to do the system overview of the recruitment tool with you I think, so let me know.

But for this endpoint we can start with constructing this new event by accessing the request data directly. If we make these fields required, the line below this comment will raise an exception, and the middleware will take care of it for us.

Reply:
Ooo ok yea could you explain more about middleware? I looked at errorWrap.js but it didn't really help haha. Also would u still need the res.json part for the status code and result or will errorWrap handle that too?

Reply:
I'd skim [this](https://expressjs.com/en/guide/using-middleware.html) real quick.

Basically in [`app.js`](https://github.com/hack4impact-uiuc/h4i-recruitment/blob/master/backend/src/app.js) it sets up the express app using a bunch of `app.use` statements, which define what happens when an endpoint is called (in the order you want them in). In particular, you can see the routes are used near the bottom (these are the specific endpoints) and then afterwards, an [`errorHandler`](https://github.com/hack4impact-uiuc/h4i-recruitment/blob/master/backend/src/middleware/errorHandler.js) is registered. This middleware simply takes an error and logs it. Basically every middleware is a function of a few things including the request, the response, and the next middleware in the chain. Normally when an error is raised, it just breaks the chain and finishes. With the errorWrap, if the endpoint throws an error, it catches it and call the `next` function, which is the `errorHandler` since that was registered directly after. That's a lot, if its unclear I could explain it on a call, it shouldn't take too long.

BUT, you don't need to worry too much about it for this fix specifically, though it is good to understand in general. All you need to know is that your code is basically wrapped in a try catch loop so that if any errors are raised, they are caught and logged. You should still check for more complex errors manually, but for simple things like a field not being set in the request, in this case you should be able to ignore it and let it raise an error. This assumes that you've made every field in your `Event` model required, otherwise an error won't even be raised.